web/tests/test_user_model.py

```diff
@@ -13,46 +14,126 @@ def test_no_password_getter():
         u.password
 
 
-def test_password_verification():
+def test_verify_password():
     u = User(password='cat')
+    assert u.password_hash is not None
     assert u.verify_password('cat')
     assert u.verify_password('dog') is False
 
 
-def test_password_salts_are_random():
-    u = User(password='cat')
-    u2 = User(password='cat')
-    assert u.password_hash != u2.password_hash
-
+def test_reset_password(session):
+    """Test the process of resetting password.
 
-def test_valid_email_change(session):
-    u = User(email='test@example.com')
+    Test wheter the password actually becomes changed when reset_password()
+    is called.The process requires that generate_reset_token is working
+    properly.
+    """
+    u = User(password='supersecret')
     session.add(u)
     session.commit()
 
-    token = u.generate_email_change_token('new-email@example.com')
-    assert u.change_email(token)
+    token = u.generate_reset_token()
+    assert u.reset_password(token, "xyz")  # You should be able to reset the password
+    assert u.verify_password("xyz")  # Checks wheter the password was changed properly
+    assert u.reset_password("fail", "xyz") is False
 
 
-def test_duplicate_email_change(session):
+def test_change_email(session):
+    """Test the process of changing email.
+
+    Test wheter the email actually becomes changed when change_email()
+    is called.The process requires that generate_email_change_token is working
+    properly.
+    """
     u = User(email='test@example.com')
     u2 = User(email='old-email@example.com')
     session.add(u)
     session.add(u2)
     session.commit()
 
-    token = u2.generate_email_change_token('test@example.com')
-    # It's not possible to change to another user's email
-    assert u2.change_email(token) is False
+    token = u.generate_email_change_token('new-email@example.com')
+    assert u.change_email(token)  # Checks wether it was possible to change email
+    assert u.email == 'new-email@example.com'  # Checks wether the new email is correct
+    assert u.change_email("xyz") is False
+    assert u2.change_email(token) is False  # It's not possible to change to another user's email!
 
 
 def test_invalid_email_change_token(session):
+    """Test wheter it is impossible to change another users email."""
     u1 = User(email='john@example.com', password='cat')
     u2 = User(email='susan@example.org', password='dog')
     session.add(u1)
     session.add(u2)
     session.commit()
+
     token = u1.generate_email_change_token('david@example.net')
-    # It's not possible to use another user's token
-    assert u2.change_email(token) is False
+    assert u2.change_email(token) is False  # It's not possible to use another user's token
     assert u2.email == 'susan@example.org'
+
+# def test_generate_email_login_token(session):
```

Comment:
It is surely okay, but I personally don't think that 29 lines with # in the beginning does not look so elegant :)